packages/ra-core/src/controller/list/filter/AutoSubmitFilterForm.tsx

```diff
@@ -15,30 +15,37 @@ import {
     ValidateForm,
 } from '../../../form';
 import { useDebouncedEvent, useEvent } from '../../../util';
-import { useListFilterContext } from '../useListFilterContext';
+import { useListContext } from '../useListContext';
 
 /**
  * TODO
  */
 export const AutoSubmitFilterForm = (props: AutoSubmitFilterFormProps) => {
-    const { filterValues, setFilters } = useListFilterContext();
+    const { filterValues, setFilters } = useListContext();
     const resource = useResourceContext(props);
 
-    const { debounce = 500, resolver, validate, children, ...rest } = props;
+    const {
+        debounce = 500,
+        resolver,
+        validate,
+        children,
+        component: Component,
```

Comment:
Why don't you provide a default value here and avoid the ternary at L124?

Reply:
I didn't know how to do it, but now I know! Thanks!